### DISCUSSION TITLE
Use input_number-entity

### DISCUSSION THREAD

**Discussion by KayGundhardt** [Disane87/spoolman-homeassistant#115] [Ideas]
Hi! Congrats to this awful HA integration!! :) What about changing the spools weight sensor-entity into an input_number-entity? This way HA automations could write information into the input_number entity (e.g. get the actual "live mass" from a scale under a reel).

**Comment by Disane87** [comment]
Thank you for this feature request!

Could you please write more about that feature request? I would know how would your workflow will be.

Should the values be saved into the appropriate spool in spoolman or only in HA?

**Comment by KayGundhardt** [comment]
Hi Marco! 

Sure, of course: all my filament rolls are each on an HX711 load cell which are connected to the HA via several ESPs (example pic can be found below). The empty roll weight and the other dead weight are now subtracted from the measured mass and then converted and written into the "pure roll mass" (an input_number-entity) using an HA automation (yaml see below). If your plugin could now write this "live mass" to the spoolman, there would be no need to manual enter the values anymore... 

I hope I was able to explain it in an understandable way? :))

```
- service: input_number.set_value
  entity_id: input_number.pla_schwarz_form_futura_masse
  data_template:
    value: >-
      {{ ( states('sensor.scale7_masse') | float) - (
      states('input_number.pla_schwarz_form_futura_leerrollenmasse') | float)}}
```

![scale7](https://github.com/Disane87/spoolman-homeassistant/assets/69385168/40101a0e-a817-4f0c-9a95-329aecbc4cb9)

**Reply by Disane87** [reply]
Neat setup! I really love the usage of the esps and the load cell. Great work! 

Thank you for your explanation. I will think about this and find a way to implement this without scrambling up all created entities. 

Actually I’m pretty torn about this since this is a pretty unusual use case. But let me sleep on it for a few nights. Can’t promise if I’m implementing this at the moment. 

besides that: do you use some Klipper compatible printer? If yes you can automatically report the spool usage to spoolman from Klipper.

**Comment by KayGundhardt** [comment]
Sure - I'm grateful for anything you can do; and maybe that's just sleeping on it. In any case, I'm glad you're taking the time to think about it! 

:)

And as for Klipper: my MK3 runs his old marlin :) I don't dare to experiment...

**Comment by ChristophCaina** [comment]
Hi :)

I am currently working on a filament scale with ESPHome, too - but right now, I am using direct API calls to Spoolman - to send the measured weight of the Spool.

Having the ability to send the new Spool weight directly via HomeAssistant would make the ESP code much simpler...
But also, I don't think that the sensor itself should be changed.

From an Architectural pov - the sensor is the correct entity, since it is only providing the information Spoolman does deliver.
But instead, you could consider to create a Service, similar as the "Use Spool filament".

Instead of sending the length or weight that was used, you should send the total weight that was measured by the scale.
This can then be used by anyone to update the information in Spoolman directly from their HA

**Reply by ChristophCaina** [reply]
just noticed, that this can already be done via the "patch spool" script / action by sending the id of the spool and the remaining weight
